panic: runtime error: invalid memory address or nil pointer dereference

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x6a3ca7]

goroutine 201 [running]:

github.com/lucas-clemente/quic-go.(*streamFramer).maybePopNormalFrames(0xc4204c2620, 0x49b, 0x0, 0x0, 0x0)
	...vendor/src/github.com/lucas-clemente/quic-go/stream_framer.go:121 +0x1a7
github.com/lucas-clemente/quic-go.(*streamFramer).PopStreamFrames(0xc4204c2620, 0x49b, 0xc42003fa80, 0x411a57, 0xc4203e8bd0)
	...vendor/src/github.com/lucas-clemente/quic-go/stream_framer.go:57 +0x91
github.com/lucas-clemente/quic-go.(*packetPacker).composeNextPacket(0xc4200b8480, 0x499, 0xc451303301, 0xb, 0x0, 0x0, 0xc42003fbc0, 0xc42003fbb8)
	...vendor/src/github.com/lucas-clemente/quic-go/packet_packer.go:269 +0x312
github.com/lucas-clemente/quic-go.(*packetPacker).PackPacket(0xc4200b8480, 0x0, 0x0, 0x0)
	...vendor/src/github.com/lucas-clemente/quic-go/packet_packer.go:159 +0x1a5
github.com/lucas-clemente/quic-go.(*session).sendPacket(0xc420238000, 0x1, 0x0, 0x0)
	...vendor/src/github.com/lucas-clemente/quic-go/session.go:826 +0x605
github.com/lucas-clemente/quic-go.(*session).run(0xc420238000, 0x0, 0x0)
	...vendor/src/github.com/lucas-clemente/quic-go/session.go:430 +0x11a
github.com/lucas-clemente/quic-go.(*server).runHandshakeAndSession.func1(0x98d840, 0xc420238000, 0xc4200ac000, 0x92a386ac494a977e)
	...vendor/src/github.com/lucas-clemente/quic-go/server.go:388 +0x34
created by github.com/lucas-clemente/quic-go.(*server).runHandshakeAndSession
	...vendor/src/github.com/lucas-clemente/quic-go/server.go:387 +0x61
```

--- a/stream_framer.go
+++ b/stream_framer.go
@@ -113,8 +113,11 @@ func (f *streamFramer) maybePopNormalFrames(maxTotalLen protocol.ByteCount) []*w
 		}
 		id := f.streamQueue[0]
 		f.streamQueue = f.streamQueue[1:]
+		// This should never return an error. Better check it anyway.
+		// The stream will only be in the streamQueue, if it enqueued itself there.
 		str, err := f.streamGetter.GetOrOpenSendStream(id)
-		if err != nil { // can happen if the stream completed after it said it had data
+		// The stream can be nil if it completed after it said it had data.
+		if str == nil || err != nil {
 			delete(f.activeStreams, id)
 			continue
 		}

--- a/stream_framer_test.go
+++ b/stream_framer_test.go
@@ -2,7 +2,6 @@ package quic
 
 import (
 	"bytes"
-	"errors"
 
 	"github.com/golang/mock/gomock"
 
@@ -139,7 +138,7 @@ var _ = Describe("Stream Framer", func() {
 		})
 
 		It("skips a stream that was reported active, but was completed shortly after", func() {
-			streamGetter.EXPECT().GetOrOpenSendStream(id1).Return(nil, errors.New("stream was already deleted"))
+			streamGetter.EXPECT().GetOrOpenSendStream(id1).Return(nil, nil)
 			streamGetter.EXPECT().GetOrOpenSendStream(id2).Return(stream2, nil)
 			f := &wire.StreamFrame{
 				StreamID: id2,

